yarpl/include/yarpl/flowable/Subscribers.h

```diff
@@ -64,38 +64,36 @@ class Subscribers {
 
  private:
   template <typename T, typename Next>
-  class Base : public InternalSubscriber<T> {
+  class Base : public BaseSubscriber<T> {
    public:
     Base(Next next, int64_t batch)
         : next_(std::move(next)), batch_(batch), pending_(0) {}
 
-    void onSubscribe(Reference<Subscription> subscription) override {
-      InternalSubscriber<T>::onSubscribe(subscription);
+    void onSubscribeImpl() override final {
       pending_ += batch_;
-      subscription->request(batch_);
+      this->request(batch_);
     }
 
-    void onNext(T value) override {
-      if (!InternalSubscriber<T>::subscription()) {
-        return;
-      }
+    void onNextImpl(T value) override final {
       try {
         next_(std::move(value));
       } catch (const std::exception& exn) {
-        InternalSubscriber<T>::subscription()->cancel();
-        onError(folly::exception_wrapper{std::current_exception(), exn});
+        this->cancel();
+        auto ew = folly::exception_wrapper{std::current_exception(), exn};
+        LOG(ERROR) << "'next' method should not throw: " << ew.what();
+        onErrorImpl(ew);
         return;
       }
 
       if (--pending_ < batch_ / 2) {
         const auto delta = batch_ - pending_;
         pending_ += delta;
-        InternalSubscriber<T>::subscription()->request(delta);
+        this->request(delta);
       }
     }
 
-   protected:
-    using InternalSubscriber<T>::onError;
+    void onCompleteImpl() override {}
```

Comment:
these should be protected, and we will not need them if we implemented them this way in the parent class.

Reply:
A future diff can improve problems that aren't introduced here, let's keep this diff focused just on removing InternalSubscriber.